machine.go

```diff
@@ -68,14 +68,33 @@ func RegisterBackend(backend Backend) {
 }
 
 func GetBackendNames() []string {
-	backendNames := make([]string, 0, len(backends))
+	backendNames := make([]string, 0, len(backends)+1)
+	backendNames = append(backendNames, "auto")
 	for backendName, _ := range backends {
 		backendNames = append(backendNames, backendName)
 	}
 	return backendNames
 }
 
 func GetBackend(name string) (Backend, error) {
+	if name == "auto" {
+		// select kvm first
+		kvm, kvm_err := GetBackend("kvm")
+		if kvm_err == nil {
+			return kvm, nil
+		}
+
+		// falling back to uml
+		uml, uml_err := GetBackend("uml")
+		if uml_err == nil {
+			return uml, nil
+		}
+
+		// no backends upported
+		return nil, fmt.Errorf("%v, %v", kvm_err, uml_err)
```

Comment:
Will this result in a meaningful error for the user ?

Reply:
here's the user-facing errors when the backends are unsupported (i expect to implement the same for debos):

```
$ fakemachine
fakemachine: Backend not supported: kvm not supported: stat /dev/kvm: no such file or directory, uml not supported: slirp-helper not installed

$ fakemachine --backend kvm
fakemachine: Backend not supported: kvm not supported: stat /dev/kvm: no such file or directory

$ fakemachine --backend uml
fakemachine: Backend not supported: uml not supported: slirp-helper not installed
```